content/docs/2.12/operate/events.md

```diff
@@ -28,3 +28,48 @@ KEDA emits the following [Kubernetes Events](https://kubernetes.io/docs/referenc
 | `TriggerAuthenticationDeleted`        | `Normal`  | When a TriggerAuthentication is deleted                                                                                     |
 | `ClusterTriggerAuthenticationAdded`   | `Normal`  | When a new ClusterTriggerAuthentication is added                                                                            |
 | `ClusterTriggerAuthenticationDeleted` | `Normal`  | When a ClusterTriggerAuthentication is deleted                                                                              |
+
+## CloudEvent Support
+
+### CloudEvent Resource
+`CloudEvent` resource now can be created in KEDA for emitting events to user's custom CloudEvent sink. Both Kubernetes Events and CloudEvents will be emitted if CloudEvent resource is created. This specification describes the `CloudEvent` Custom Resource definition:
+
+[`cloudevent_types.go`](https://github.com/kedacore/keda/blob/v1.4.0/pkg/apis/keda/v1alpha1/cloudevent_types.go)
+
+```yaml
+apiVersion: keda.k8s.io/v1alpha1
+kind: CloudEvent
+metadata:
+  name: {cloud-event-name}
+spec:
+  clusterName: {cluster-name} #Optional. Will be used in source/subject 
+  destination:
+    http:
+      uri: http://foo.bar
+```
+
+In general, an event emitted by KEDA would fundamentally come down to the following structure:
+```json
+{
+    "specversion" : "1.0",
+    "type" : "com.cloudevents.keda",
+    "source" : "/{cluster-name}/{namespace}/keda",
+    "subject" : "/{cluster-name}/{namespace}/workload/{scaledobject-name}",
+    "id" : "<guid>",
+    "time" : "2018-04-05T17:31:00Z",
+    "datacontenttype" : "application/json",
+    "data" : {
+      "reason":"<event-reason>",
+      "message":"<event-message>"
+   }
+}
+```
+
+### Destination
+There will be multiple type of destination to emit KEDA events. Nowadays an HTTP CloudEvent destination is supported.
+#### CloudEvent HTTP
+```yaml
+  destination:
+    http:
+      uri: http://foo.bar  #An http endpoint that can receive cloudevent
+```
```

Comment:
As CloudEvents is a bigger feature than current cases, maybe we should anticipate it and create a section for it since now. Or maybe it's too early and we won't need it, just a non blocking suggestion. @tomkerkhove  @zroubalik ?

Reply:
I'm OK with both options